internal/datastore/postgres/migrations/driver.go

```diff
@@ -37,6 +37,14 @@ func NewAlembicPostgresDriver(ctx context.Context, url string) (*AlembicPostgres
 	pgxcommon.ConfigurePGXLogger(connConfig)
 	pgxcommon.ConfigureOTELTracer(connConfig)
 
+	if awsIamAuthEnabled {
```

Comment:
This would turn into something like `if credentialsProvider != nil  { username, password, err := credentialsProvider.Get() ...  }`...